### ISSUE TITLE
Commit/Run/BenchmarkResult/... timestamps: clarify timezone across codebase

### ISSUE THREAD

**Issue by jgehrcke** [conbench/conbench#609]
Conbench should after all be useful for globally distributed teams that operate at the same time across the various timezones of the world. We have to address timezone confusion systematically.

Let's do one of or some of or all of:
- add code comments
- turn tz-naive in tz-aware datetime objects before storing in database

 Relates to https://github.com/conbench/conbench/issues/534.

**Comment by jonkeane** [comment]
💯 on this.

an additional point (or maybe a slight addition to "turn tz-naive in tz-aware datetime objects before storing in database"). IMO we should heavily encourage folks to send timestamps that are TZ-aware, but if we _receive_ TZ-naive datetimes, we should turn those into UTC or assume UTC. That way we cover the most common cases (systems running CI somewhere where the TZ is typically UTC by default) and we're not rejecting data in cases where someone is sending something kind of off

**Comment by jgehrcke** [comment]
> "turn tz-naive in tz-aware datetime objects before storing in database")

In https://github.com/conbench/conbench/pull/614 I decided against doing so because seemingly most (all?) of the DB columns we use for storing timestamps have been created so that they do _not_ track timezone information, example: https://github.com/conbench/conbench/blob/cc291e8566d77b527c27348fe7ba97ee8d44c2f5/conbench/entities/benchmark_result.py#L47

That's OK, too, if we introduce a strict convention that those always refer to UTC. I have started doing so by adding code comments etc.

**Comment by jgehrcke** [comment]
> IMO we should heavily encourage folks to send timestamps that are TZ-aware, but if we receive TZ-naive datetimes, we should turn those into UTC or assume UTC.

Right! Did this for `finished_timestamp` for `RunCreate` in #614. The updated spec (feedback still welcome!) clarifies this behavior to users, and reads as

> A datetime string indicating the time at which the run finished. Expected to be in ISO 8601 notation. Timezone-aware notation recommended. Timezone-naive strings are interpreted in UTC. Fractions of seconds can be provided but are not returned by the API. Example value: 2022-11-25T22:02:42Z.

**Comment by jgehrcke** [comment]
Next up I am doing #614 for `BenchmarkCreate`'s `timestamp` property.

**Comment by jgehrcke** [comment]
More work for `Run.timestamp`: https://github.com/conbench/conbench/pull/647

**Comment by jgehrcke** [comment]
TODO: find out if there are any more user-facing timestamp concepts left that need clarification.
